### PR TITLE
Fix /candidates/progress/by-req/ and /candidates/progress/stats/ errors

### PR DESCRIPTION
**Pull request by JunoLee128**
Add None type checks, and fix an HTML syntax issue

### CODE REVIEW

File: candidates/views.py [hunk 1]
```diff
@@ -953,7 +953,7 @@ def get_context_data(self, **kwargs):
             for req_progress in progress_item['requirements']:
                 requirement = req_progress['requirement']
                 # Exclude progresses with the wrong requirement
-                if self.display_req and self.display_req != requirement:
+                if not requirement or self.display_req and self.display_req != requirement:
```
Comment (ochan1) [inline candidates/views.py:956]: Use the following syntax (prioritize easy to read, even though logic may be is the same if you were thinking of that)

`if (not requirement) or (self.display_req and self.display_req != requirement):`

File: candidates/views.py [hunk 2]
```diff
@@ -998,6 +998,8 @@ def get_context_data(self, **kwargs):
         total_unfinished = 0
         for progress_item in progress.values():
             for req_progress in progress_item['requirements']:
+                if not req_progress['requirement']:
+                    continue
```
Comment (ochan1) [inline candidates/views.py:1002]: Are there any implications of just "continuing" for an empty Requirement in req_progress?
Just curious and making sure you thought about this

**Review by ochan1 — CHANGES_REQUESTED** [review]
Code looks to mostly make sense
Please address the questions and recommended corrections, afterwards ready to merge in

File: candidates/views.py (hunk 2 above)
Comment (JunoLee128) [reply candidates/views.py:1002]: I saw a similar structure elsewhere in the code. It looks like the req_progress variables with an empty requirement don't have much purpose / might have been added accidentally? They probably don't need to exist

File: candidates/views.py (hunk 1 above)
Comment (JunoLee128) [reply candidates/views.py]: done

File: candidates/views.py (hunk 2 above)
Comment (ochan1) [reply candidates/views.py:1002]: Ok, yeah now that you mentioned it
When I investigated the issue, the specific requirement was not filled in which resulted in the None that was included in the dictionary

Agreed and OK to skip it then

**Review by ochan1 — APPROVED** [review]
Corrections addressed
LGTM

**Pull request merged**